src/core/qgsvectorlayerexporter.cpp

```diff
@@ -398,6 +411,27 @@ QgsVectorLayerExporter::exportLayer( QgsVectorLayer *layer,
         return ErrProjection;
       }
     }
+
+    // Handles conversion to single-part
+    if ( convertGeometryToSinglePart && fet.geometry().isMultipart() )
+    {
+      QgsGeometry singlePartGeometry { fet.geometry() };
+      // We want a failure if the geometry cannot be converted to single-part without data loss!
+      // check if there are more than one part
+      const QgsGeometryCollection *c = qgsgeometry_cast<QgsGeometryCollection *>( singlePartGeometry.constGet() );
```

Comment:
```suggestion
      const QgsGeometryCollection *c = qgsgeometry_cast<const QgsGeometryCollection *>( singlePartGeometry.constGet() );
```

Hmm, this really shouldn't compile without the const...